command/registry.go

```diff
@@ -0,0 +1,176 @@
+package command
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/packer/hcl2template"
+	"github.com/hashicorp/packer/internal/registry"
+	"github.com/hashicorp/packer/internal/registry/env"
+	"github.com/hashicorp/packer/packer"
+
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+const (
+	// Known HCP Packer Image Datasource, whose id is the SourceImageId for some build.
+	hcpImageDatasouceType string = "hcp-packer-image"
+	buildLabel            string = "build"
+)
+
+// SetupRegistry attempts to setup the HCP-related structures if HCP is enabled
+// for the command
+func SetupRegistry(cfg packer.Handler) hcl.Diagnostics {
+	switch cfg := cfg.(type) {
+	case *hcl2template.PackerConfig:
+		return setupRegistryForPackerConfig(cfg)
+	case *CoreWrapper:
+		return setupRegistryForPackerCore(cfg)
+	}
+
+	panic("NOPE.")
+}
+
+func setupRegistryForPackerConfig(pc *hcl2template.PackerConfig) hcl.Diagnostics {
+	var diags hcl.Diagnostics
+
+	hasHCP := false
+	for _, build := range pc.Builds {
```

Comment:
Based off this I'm thinking there can be a situation where we have a template with multiple build blocks with one or more having an hcp_packer_registry block defined. Is that correct? 

still thinking about this change a bit.

Reply:
With the last reroll this isn't possible anymore and the current behaviour of when multiple build blocks are defined in HCP mode it errors is replicated.